### ISSUE TITLE
custom validation is not working when the value in field is a zero-value 

### ISSUE THREAD

**Issue by muly** [asaskevich/govalidator#130]
Scenario: I have a struct that I'm validating. for the reason that i'm giving towards the end, I had to create custom validations to check if "the field" has a non-zero value. but for some reason, the custom validation function is not executing when the value in "the field" is a zero value. I have also added a print statement to check if the function is fired or not. for record which has a non-zero value in "the field", i see the message in the print statement being printed but not in case of the records with zero value in "the field".

I know that we can use "required" tag for this validation, but the reason for me to create a custom tag instead is; the required rule is not static. the value in "the field" is required only for some records but not the others based on the value of "the other field". I implemented the logic to create and cleanup custom validation tag for each record based on the value in "the other field", this is good. but i'm not able to figureout why the custom validation is not firing when the value in "the field" is a zero value.

I hope that the description of the issue is clear. Please let me know if you need an example data and code snippet.

**Comment by annismckenzie** [comment]
A quick example would be great, a failing unit test would be even better – you can find the custom validation tests here: [validator_test.go#L1924](/asaskevich/govalidator/blob/master/validator_test.go#L1924). When #123 lands you'll also have access to the struct being validated in the custom validation function; that should cut down on the amount of code necessary for your scenario.

Now, I don't _really_ get what you're saying. The custom validation function runs when it's defined on a field and you are responsible for returning true or false all on your own. If there's some condition where a zero-value is acceptable then return true in that instance. Required or optional validation tags are ignored when custom validators are defined on a field. I don't really see why the custom validation wouldn't be triggered. Are you using a custom validation function defined inline and access closure variables/objects from the outside to figure out whether to validate or not?

**Comment by OlgaDnepr** [comment]
Hi,

I've got the same error. My custom validator doesn't validate empty fields.  I believe that the problem is in line **813** in **validator.go** file:
```
	if isEmptyValue(v) {
		// an empty value is not validated, check only required
		return checkRequired(v, t, options)
	}
```

**Comment by geedchin** [comment]
@OlgaDnepr hi, how did you  deal with the issue?

**Comment by asaskevich** [comment]
Thank you for your investigation, it seems to be fixed in #329.
Sorry for long awaited response.
